src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -258,6 +263,368 @@ public static async Task LargeJsonFile(int bufferSize)
                 Assert.Equal(SessionResponseCount, deserializedList.Count);
             }
         }
+
+        [Theory]
+//      [InlineData(1, true, true)]   ActiveIssue https://github.com/dotnet/corefx/issues/41598
+//      [InlineData(1, true, false)]
+        [InlineData(1, false, true)]
+        [InlineData(1, false, false)]
+//      [InlineData(10, true, false)]
+        [InlineData(10, false, false)]
+        [InlineData(100, false, false)]
+        [InlineData(1000, false, false)]
+        public static async Task VeryLargeJsonFileTest(int payloadSize, bool ignoreNull, bool writeIndented)
+        {
+            List<Order> list = PopulateLargeObject(payloadSize);
+
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                IgnoreNullValues = ignoreNull,
+                WriteIndented = writeIndented
+            };
+
+            string json = JsonSerializer.Serialize(list, options);
+
+            // Sync case.
+            {
+                List<Order> deserializedList = JsonSerializer.Deserialize<List<Order>>(json, options);
+                Assert.Equal(payloadSize, deserializedList.Count);
+
+                string jsonSerialized = JsonSerializer.Serialize(deserializedList, options);
+                Assert.Equal(json, jsonSerialized);
+            }
+
+            // Async case.
+            using (var memoryStream = new MemoryStream())
+            {
+                await JsonSerializer.SerializeAsync(memoryStream, list, options);
+                string jsonSerialized = Encoding.UTF8.GetString(memoryStream.ToArray());
+                Assert.Equal(json, jsonSerialized);
+
+                memoryStream.Position = 0;
+                List<Order> deserializedList = await JsonSerializer.DeserializeAsync<List<Order>>(memoryStream, options);
+                Assert.Equal(payloadSize, deserializedList.Count);
+            }
+        }
+
+        [Theory]
+/*        [InlineData(1, true, true)] ActiveIssue https://github.com/dotnet/corefx/issues/41598
```

Comment:
@layomia - can you verify that these tests pass once your PR is in that fixes https://github.com/dotnet/corefx/issues/41598 (i.e. https://github.com/dotnet/corefx/issues/41598).